### DISCUSSION TITLE
Custom Simplify Constant Rule Exceptions?

### DISCUSSION THREAD

**Discussion by calvbore** [josdejong/mathjs#3273] [Q&A]
I'd like to specify simplification exceptions. What would be the easiest way to go about this?

Right now I'm getting the derivatives of log functions like this:
```
derivative('log(x, 10)', 'x').toString(); // outputs 0.43429448190325176 / x
derivative('log(x, 10, {simplify: false})', 'x').toString(); // 1 / x * log(8)
```
I would like to keep the output for a function on a constant in the form of the second line. However, if I use the option `{simplify: false}` on a more complex derivative the output then becomes quite messy. I would like for everything else except for the constant log function to be simplified. I know that I can use custom rules in the `simplify` function. I'm wondering how to implement a rule with the desired behaviour?

The solution that I'm thinking so far is to copy the  `simplify.rules` array and then replace the instances of `simplifyConstant` with a function that wraps it and only runs `simplifyConstant` if it is not a log function. Is there another, or better, way to accomplish this? Is this a completely misguided approach?

(Additionally, what would be the best way filter if a node is a log function?) 

Thanks!

**Comment by josdejong** [comment]
The best way is indeed to take `simplify.rules` and add/remove rules to your liking. You can create a rule yourself, define the rule as a function that receives a `node`. You can check whether the node is a log function like:

```js
function myCustomSymplifyRule(node) {
  if (
    node.isFunctionNode && 
    node.fn.isSymbolNode && 
    node.fn.name === 'log'
  ) {
    // ...
  }

  return node
}
```
